src/widgetastic/widget.py

```diff
@@ -1173,6 +1173,10 @@ def fill(self, value):
             else:
                 self.logger.info('Filling column %r', key)
 
+            # if the row widgets aren't visible the row needs to be clicked to edit
+            if not self[key].widget.is_displayed:
+                self.click()
+                changed = True
```

Comment:
Hmm, maybe make this conditional? And the changed may be misleading because you did not check the visibility of the widget afterwards.